### PR TITLE
Adding robustica clustering plots to the TEDANA report

### PR DESCRIPTION
**Pull request by BahmanTahayori**
To visualise the clustering results created by `robustica`, I have added 2D plots using T-distributed Stochastic Neighbor Embedding (TSNE), https://scikit-learn.org/stable/modules/generated/sklearn.manifold.TSNE.html, in this pull request. In the created 2D plot each black circle represents a clustered run for an ICA component. I used dilated convex hulls to illustrate clusters for each ICA component.  When DBSCAN is used for clustering (the first option in ica.py in `tedana` 24.0.2), there may be runs considered noisy samples and thus are not clustered. These samples are labelled by -1. In a second plot, these unclustered runs are superimposed on the first plot and are distinguished by red crosses. These plots are saved in the _figures_ folder of the subject where other generated figures are saved.  Examples of these plots are presented here. I have not included any numbers on the axes as the numbers do not convey meaningful information. Of note, if there is no cluster labelled -1, only one figure is generated. 

clustering_projection_tsne
clustering_projection_tsne_with_noise

These plots and the Index Quality (IQ) provide an insight into the quality of the `robustica` results. The objective is to add these plots (interactively) to the HTML report. 

@Lestropie, could you please look at this internal PR and send me your feedback?

### CODE REVIEW

File: tedana/reporting/static_figures.py [hunk 1]
```diff
@@ -798,3 +800,102 @@ def plot_adaptive_mask(
...
+        An array containing the coordinates of projected data.
+    io_generator : object
+        An object containing all the information needed to generate the output.
+    """
+    title = "2D projection of clustered ICA runs using TSNE"
+    line_width = 0.6
+    marker_size = 3
+    convex_hull_color = "b--"
+    point_color = "k"
+    dilation_factor = 1.5
```
Comment (Lestropie) [inline tedana/reporting/static_figures.py:822]: These (and potentially other plotting variables later in the function) could be defined as constants at the head of the file, to distinguish them from source code.

**Comment by Lestropie** [comment]
I can't comment a great deal on the code responsible for generating the plots, but some thoughts just from looking at the figures presented:

-   If in the presence of unclustered components you are generating two plots, one omitting those from the plot and once including them, I would remove the former, as it is misleading with respect to the data. Just produce a single plot and show all unclustered components, even if the number of unclustered components is zero.

-   Is the purpose of the dilation of the convex hull to ensure that the individual component dots remain visible? Unless there is a strong justification for the use of such dilation in visualisations of this type, I would err toward not doing that dilation, as it is a distortion of the actual data, particularly if there are two clusters quite close to one another and one is interested in the grouping vs. separation of those clusters. To mitigate the visual clutter of showing the convex hull lines passing through the component points you could try using two more distinct colours, or having the convex hull lines partially transparent, or having larger gaps in the dashes.
    Or alternatively you could try *filling* each convex hull rather than drawing the outline. Here a colour very close to the background could be used and it would still be possible to discriminate such clusters from the surrounding empty space, but the component black dots would have a strong contrast against both.
    Is it mathematically possible to have two clusters overlap in a plot such as this? They may be discriminable in higher dimensions but overlap in this 2D projection. In this case having dilated convex hulls becomes a problem as it's distorting the magnitude of that overlap. If this were done using a transparent fill, then those fills should stack and the overlap should become more prominent.

-   Including all relevant information about what is presented in the plot, including the clustering method and the value of IQ, somewhere in the plot (eg. title), would be beneficial.

-   I would use more verbose axis labels than "`x1`" and "`x2`".

-   In the legend, should "`runs`" be "`components`"? Ie. Is it a specific component of a specific ICA run that is unclustered, or is it the ICA run in its entirety?
